### ISSUE TITLE
Question about the performance of pre-trained model that the link contains. 

### ISSUE THREAD

**Issue by Zhou2019** [wp03052/MATRN#5]
First, thank you very much for your work, it is very impressive. However, when I evaluate with the pre-trained model provided by the link, I get results that are lower than the performance of the report. What is the reason for this? Thank you very much for your answer.
And my results on 6 datasets of IIIT5k_3000, SVT, SVTP, IC13_857, IC15_1811, CUTE80 are as follows:

[2022-03-03 23:28:19,374 main.py:276 INFO train-matrn] validation time: 62.44528245925903 / batch size: 384
[2022-03-03 23:28:19,374 main.py:281 INFO train-matrn] eval loss =  1.435,  ccr =  0.957,  cwr =  0.904,  ted = 1542.000,  ned =    297,  ted/w =  0.213.

you results of the same six datasets average cwr is 93.450.

Thank you very much again!

**Comment by byeonghu-na** [comment]
Thank you for your interest.
I evaluate our model with the provided link, and I reproduce reported performances.

~~~
[2022-03-04 11:55:08,553 main.py:255 INFO train-matrn] Construct dataset.
[2022-03-04 11:55:08,557 main.py:94 INFO train-matrn] 7248 training items found.
[2022-03-04 11:55:08,557 main.py:96 INFO train-matrn] 7248 valid items found.
[2022-03-04 11:55:08,557 main.py:97 INFO train-matrn] 6992 test items found.
[2022-03-04 11:55:08,557 main.py:259 INFO train-matrn] Construct model.
[2022-03-04 11:55:08,840 model_vision.py:39 INFO train-matrn] Read vision model from ./results/pretrain/pretrain-vision-model/best-pretrain-vision-model.pth.
[2022-03-04 11:55:12,497 model_language.py:38 INFO train-matrn] Read language model from ./results/pretrain/pretrain-language-model/pretrain-language-model.pth.
[2022-03-04 11:55:12,589 main.py:109 INFO train-matrn] num_params: 44158489
[2022-03-04 11:55:12,589 main.py:262 INFO train-matrn] Construct learner.
[2022-03-04 11:55:12,757 main.py:168 INFO train-matrn] Use 4 GPUs.
[2022-03-04 11:55:13,319 main.py:185 INFO train-matrn] Read model from best-train-matrn
[2022-03-04 11:55:13,319 main.py:271 INFO train-matrn] Start validate
[2022-03-04 11:55:40,190 main.py:275 INFO train-matrn] validation time: 26.870068788528442 / batch size: 384
[2022-03-04 11:55:40,190 main.py:280 INFO train-matrn] eval loss =  1.509,  ccr =  0.968,  cwr =  0.935,  ted = 1065.000,  ned =    217,  ted/w =  0.147,
~~~

Please check once again (dataset, checkpoint, ...), and if you have same result, then could you provide full result and your environment?

Thank you.

**Comment by Zhou2019** [comment]
Thank you very much for your reply. I checked my dataset and other configurations again and tested several times, but the results are still not as good as in the paper, and the results always fluctuate each time I test. Here is my environment configuration and complete results.

This is my configuration file. [train_matrn_me.txt](https://github.com/wp03052/MATRN/files/8185157/train_matrn_me.txt)
This is my environment: python3.7.9 fastai=1.0.60 torch=1.6.0 torchvision=0.7.0.
This is my complete results of several test.
  ```
[2022-03-04 16:59:21,235 main.py:256 INFO train-matrn] Construct dataset.
  [2022-03-04 16:59:21,240 main.py:95 INFO train-matrn] 7248 training items found.
  [2022-03-04 16:59:21,240 main.py:97 INFO train-matrn] 7248 valid items found.
  [2022-03-04 16:59:21,240 main.py:98 INFO train-matrn] 6992 test items found.
  [2022-03-04 16:59:21,240 main.py:260 INFO train-matrn] Construct model.
  [2022-03-04 16:59:21,584 model_vision.py:40 INFO train-matrn] Read vision model from ./../best-pretrain-vision-model.pth.
  [2022-03-04 16:59:28,778 model_language.py:39 INFO train-matrn] Read language model from ./../pretrain-language-model.pth.
  [2022-03-04 16:59:29,112 main.py:110 INFO train-matrn] num_params: 44158489
  [2022-03-04 16:59:29,112 main.py:263 INFO train-matrn] Construct learner.
  [2022-03-04 16:59:29,385 main.py:169 INFO train-matrn] Use 4 GPUs.
  [2022-03-04 16:59:32,143 main.py:183 INFO train-matrn] Read model from ../best-train-matrn.pth
  [2022-03-04 16:59:32,213 main.py:272 INFO train-matrn] Start validate
  [2022-03-04 17:00:12,832 main.py:276 INFO train-matrn] validation time: 40.61845088005066 / batch size: 384
  [2022-03-04 17:00:12,833 main.py:281 INFO train-matrn] eval loss =  1.466,  ccr =  0.956,  cwr =  0.901,  ted = 1702.000,  ned =    326,  ted/w =  0.235,
```

```
[2022-03-04 18:30:55,835 main.py:256 INFO train-matrn] Construct dataset.
[2022-03-04 18:30:55,840 main.py:95 INFO train-matrn] 7248 training items found.
[2022-03-04 18:30:55,840 main.py:97 INFO train-matrn] 7248 valid items found.
[2022-03-04 18:30:55,840 main.py:98 INFO train-matrn] 6992 test items found.
[2022-03-04 18:30:55,840 main.py:260 INFO train-matrn] Construct model.
[2022-03-04 18:30:56,181 model_vision.py:40 INFO train-matrn] Read vision model from ./../best-pretrain-vision-model.pth.
[2022-03-04 18:31:03,292 model_language.py:39 INFO train-matrn] Read language model from ./../pretrain-language-model.pth.
[2022-03-04 18:31:03,602 main.py:110 INFO train-matrn] num_params: 44158489
[2022-03-04 18:31:03,602 main.py:263 INFO train-matrn] Construct learner.
[2022-03-04 18:31:03,864 main.py:169 INFO train-matrn] Use 4 GPUs.
[2022-03-04 18:31:06,614 main.py:183 INFO train-matrn] Read model from ../best-train-matrn.pth
[2022-03-04 18:31:06,683 main.py:272 INFO train-matrn] Start validate
[2022-03-04 18:31:46,661 main.py:276 INFO train-matrn] validation time: 39.977182149887085 / batch size: 384
[2022-03-04 18:31:46,661 main.py:281 INFO train-matrn] eval loss =  1.405,  ccr =  0.958,  cwr =  0.913,  ted = 1413.000,  ned =    275,  ted/w =  0.195,
```

```
[2022-03-04 15:58:10,078 main.py:256 INFO train-matrn] Construct dataset.
[2022-03-04 15:58:10,086 main.py:95 INFO train-matrn] 7248 training items found.
[2022-03-04 15:58:10,086 main.py:97 INFO train-matrn] 7248 valid items found.
[2022-03-04 15:58:10,086 main.py:98 INFO train-matrn] 6992 test items found.
[2022-03-04 15:58:10,086 main.py:260 INFO train-matrn] Construct model.
[2022-03-04 15:58:10,576 model_vision.py:40 INFO train-matrn] Read vision model from ./../best-pretrain-vision-model.pth.
[2022-03-04 15:58:23,626 model_language.py:39 INFO train-matrn] Read language model from ./../pretrain-language-model.pth.
[2022-03-04 15:58:24,757 main.py:110 INFO train-matrn] num_params: 44158489
[2022-03-04 15:58:24,757 main.py:263 INFO train-matrn] Construct learner.
[2022-03-04 15:58:46,211 main.py:183 INFO train-matrn] Read model from ../best-train-matrn.pth
[2022-03-04 15:58:46,292 main.py:272 INFO train-matrn] Start validate
[2022-03-04 15:59:50,782 main.py:276 INFO train-matrn] validation time: 64.49005603790283 / batch size: 384
[2022-03-04 15:59:50,783 main.py:281 INFO train-matrn] eval loss =  1.470,  ccr =  0.955,  cwr =  0.902,  ted = 1593.000,  ned =    313,  ted/w =  0.220, 
```


And this is the full results of yesterday's.
```
[2022-03-03 23:26:41,767 main.py:256 INFO train-matrn] Construct dataset.
[2022-03-03 23:26:41,774 main.py:95 INFO train-matrn] 7248 training items found.
[2022-03-03 23:26:41,774 main.py:97 INFO train-matrn] 7248 valid items found.
[2022-03-03 23:26:41,775 main.py:98 INFO train-matrn] 6992 test items found.
[2022-03-03 23:26:41,775 main.py:260 INFO train-matrn] Construct model.
[2022-03-03 23:26:42,372 main.py:110 INFO train-matrn] num_params: 44158489
[2022-03-03 23:26:42,372 main.py:263 INFO train-matrn] Construct learner.
[2022-03-03 23:27:16,873 main.py:183 INFO train-matrn] Read model from ../best-train-matrn.pth
[2022-03-03 23:27:16,928 main.py:272 INFO train-matrn] Start validate
[2022-03-03 23:28:19,374 main.py:276 INFO train-matrn] validation time: 62.44528245925903 / batch size: 384
[2022-03-03 23:28:19,374 main.py:281 INFO train-matrn] eval loss =  1.435,  ccr =  0.957,  cwr =  0.904,  ted = 1542.000,  ned =    297,  ted/w =  0.213, 
```

I tried downloading the evaluation datasets from clovaai again, but it didn't work. In addition, I did not load the best-pretrain-vision-model.pth and pretrain-language-model.pth in the first test (i.e. the evaluation results at the beginning of the question), but I tried to add them later (i.e. the current results), and it did not work either. I don't know how to fix him anymore. 
What should I do please?
Thank you very much.

**Comment by byeonghu-na** [comment]
Oh, I actually modify codes today (commit 861e392) to change self.model to self.model1.
Because `semantic_visual` class has `model1` instance (not `model`) in provided checkpoint model.
But, you may not get errors since you turn off strict model load. So, `model` in `semantic_visual` class does not load parameters from the checkpoint. So, results fluctuate each time due to random initialization.

Sorry for confusion. Please check again by updating the code to the latest version.

**Comment by Zhou2019** [comment]
Oh! Yeah.
Thanks very muchhhhhhh!

I remembered, when I first debugged I encountered this error and corrected it, but somehow it went back in the later modifications, and I couldn't get rid of the error because I closed the strict model load.After fix this error,the evaluated performance is now comparable to that in the paper. Thank you very much again.
